cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -110,3 +160,37 @@ func (pe *ProbeEvent) initOrErrorEvent() {
 		klog.Error(err)
 	}
 }
+
+// generateUUID creates a new UUID based on the algorithm proposed in
+// https://github.com/openebs/openebs/pull/2666
+func generateUUID(bd blockdevice.BlockDevice) (string, bool) {
+	var ok bool
+	var uuidField, uuid string
+
+	// select the field which is to be used for generating UUID
+	switch {
+	case bd.DeviceAttributes.DeviceType == libudevwrapper.UDEV_PARTITION:
+		klog.Infof("device(%s) is a partition, using partition UUID: %s", bd.DevPath, bd.PartitionInfo.PartitionEntryUUID)
+		uuidField = bd.PartitionInfo.PartitionEntryUUID
+		ok = true
+	case len(bd.DeviceAttributes.WWN) > 0:
+		klog.Infof("device(%s) has a WWN, using WWN: %s", bd.DevPath, bd.DeviceAttributes.WWN)
+		uuidField = bd.DeviceAttributes.WWN
+		ok = true
+	case len(bd.PartitionInfo.PartitionTableUUID) > 0:
```

Comment:
Can you add a comment on how this case is different from the first case at line 172?

Reply:
Comments added.